lib/scope.js

```diff
@@ -120,7 +120,73 @@ module.exports = {
     expect( elem.constructor.name, `Did not find anything where the ${ ordinal } "${ label_text }" ${ roles } should have been. ${ elem.constructor.name }` ).to.equal('ElementHandle');
 
     return elem;
-  },  // Ends getSpecifiedChoice
+  },  // Ends getSpecifiedChoice()
+
+  tapElement: async function tapElement(scope, elem) {
+    /* Tap a given element, detect navigation, and do appropriate related things. */
+    expect( elem, 'That element cannot be found, so it cannot be tapped.' ).to.exist;
+
+    let start_url = await scope.page.url()
+
+    let winner;
+    if (elem) {
+      winner = await Promise.race([
+        Promise.all([  // Error loads page, so no need to detect
+          // Click with no navigation will end immediately
+          elem[ scope.activate ](),
+          scope.page.waitForNavigation({waitUntil: 'domcontentloaded'}),
+        ]),
+        scope.page.waitForSelector('.alert-danger'),
+        scope.page.waitForSelector('.da-has-error'),
```

Comment:
comment - this is not meant to detect breaking errors, just user-visible errors for user to fix